crates/router/src/core/routing/helpers.rs

```diff
@@ -909,34 +913,313 @@ pub fn generate_tenant_business_profile_id(
+                business_profile,
+                feature_to_enable,
+                dynamic_routing_algo_ref,
+                dynamic_routing_type,
+                dynamic_routing.success_based_algorithm,
+            )
+            .await
+        }
+        routing_types::DynamicRoutingType::EliminationRouting => {
+            enable_specific_routing_algorithm(
+                state,
+                key_store,
+                business_profile,
+                feature_to_enable,
+                dynamic_routing_algo_ref,
+                dynamic_routing_type,
+                dynamic_routing.elimination_routing_algorithm,
+            )
+            .await
+        }
+    }
+}
+
+#[cfg(all(feature = "v1", feature = "dynamic_routing"))]
+pub async fn enable_specific_routing_algorithm<A>(
+    state: &SessionState,
+    key_store: domain::MerchantKeyStore,
+    business_profile: domain::Profile,
+    feature_to_enable: routing_types::DynamicRoutingFeatures,
+    mut dynamic_routing_algo_ref: routing_types::DynamicRoutingAlgorithmRef,
+    dynamic_routing_type: routing_types::DynamicRoutingType,
+    algo_type: Option<A>,
+) -> RouterResult<ApplicationResponse<routing_types::RoutingDictionaryRecord>>
+where
+    A: routing_types::DynamicRoutingAlgoAccessor + Clone + std::fmt::Debug,
+{
+    // Algorithm wasn't created yet
+    let Some(mut algo_type) = algo_type else {
+        return default_specific_dynamic_routing_setup(
+            state,
+            key_store,
+            business_profile,
+            feature_to_enable,
+            dynamic_routing_algo_ref,
+            dynamic_routing_type,
+        )
+        .await;
+    };
+
+    // Algorithm was in disabled state
+    let Some(algo_type_algorithm_id) = algo_type
+        .clone()
+        .get_algorithm_id_with_timestamp()
+        .algorithm_id
+    else {
+        return default_specific_dynamic_routing_setup(
+            state,
+            key_store,
+            business_profile,
+            feature_to_enable,
+            dynamic_routing_algo_ref,
+            dynamic_routing_type,
+        )
+        .await;
+    };
+    let db = state.store.as_ref();
+    let profile_id = business_profile.get_id().clone();
+    let business_profile = business_profile.clone();
+    let algo_type_enabled_features = algo_type.get_enabled_features();
+    if *algo_type_enabled_features == feature_to_enable {
+        // algorithm already has the required feature
+        return Err(errors::ApiErrorResponse::PreconditionFailed {
+            message: format!("{} is already enabled", dynamic_routing_type),
+        }
+        .into());
+    };
+    *algo_type_enabled_features = feature_to_enable.clone();
+    dynamic_routing_algo_ref
+        .update_specific_ref(dynamic_routing_type.clone(), feature_to_enable.clone());
+    update_business_profile_active_dynamic_algorithm_ref(
+        db,
+        &state.into(),
+        &key_store,
+        business_profile.clone(),
```

Comment:
clone not required. I see that you are using `business_profile` below to get `profile_id`. Store profile_id alone in a variable before this call and move `business_profile`